### PR TITLE
Fix tensor label type inference in default collator

### PR DESCRIPTION
**Pull request by joeddav**
Quick fix to `default_data_collator` allowing it to recognize the correct type of PT tensor label inputs rather than always casting them to float (#5060)

### CODE REVIEW

**Comment by julien-c** [comment]
What's the perf impact of a try/catch?

**Comment by joeddav** [comment]
Is it supposed to be slower? I ran a quick profile iterating over IMDb train set and saw no difference, but can manually check for `torch.tensor` instead if that's preferred.

**Review by LysandreJik — APPROVED** [review]
Cool!
